Realm/RLMObjectStore.mm

```diff
@@ -202,10 +210,36 @@ static bool RLMRealmGetTables(RLMRealm *realm, RLMSchema *targetSchema) {
 static bool RLMPropertyHasChanged(RLMProperty *p1, RLMProperty *p2) {
     return p2 == nil
         || p1.type != p2.type
+        || p1.optional != p2.optional
         || ![p1.name isEqualToString:p2.name]
         || (p1.objectClassName != p2.objectClassName && ![p1.objectClassName isEqualToString:p2.objectClassName]);
 }
 
+static bool RLMPropertyCanBeMigrated(RLMProperty *oldProperty, RLMProperty *newProperty) {
```

Comment:
I was envisioning we could add other auto-migrations, like Date -> Double, Int -> Float, etc?